sg14-error-codes.html

```diff
@@ -567,6 +595,12 @@
     are "special," i.e., they will not be treated as "errors" by any of the machinery in the Standard. Error codes with
     value <code>0</code> are effectively reserved for the "success" case, and programmers should not attempt to use them
     for any other purpose.
+    
+    <!-- ned: I object in the strongest terms to the standard doing any such thing. Code = 0 is a perfectly valid failure code.
+    system_error should NOT be forcing this onto people. What should be done instead is a new constexpr null_category
+    and the if(ec) boolean test should be implemented as return *this == std::error_code(); where a default constructed
+    error code has code zero with the null_category. These will be 100% trivial types, there is no cost to requiring comparison
+    to a constexpr constructed default instance -->
```

Comment:
I will soften the "SG14 would like..." wording. I actually agree with your position here and would like to work it in either here in 4.3 or else as a new section 4.7 specifically about null_category and default-constructibility.

Do you also think that `std::error_condition` should be constexpr-default-constructible and receive the null_category? Right now `error_code` gets `{0, system_category()}` and `error_condition` gets `{0, generic_category()}`, which means that [it is implementation-defined and in practice vendors differ](https://wandbox.org/permlink/I8WlkzPgvixsuMM8) as to whether

    std::error_code{} == std::error_condition{}

(libstdc++ and MSVC both say no; libc++ says yes).

Reply:
I just followed up on the SG14 reflector, before seeing this comment. Thinking about it, given the "`error_condition` is just for comparing" (which I acknowledge is not universally held, but comes from the original write-ups for Boost::asio), having 0 be reserved for all `error_condition`s as 'success' might still be beneficial. Although I also realise that this might not work with @ned14 's comment earlier that an enum can be both `error_code` and `error_condition`.